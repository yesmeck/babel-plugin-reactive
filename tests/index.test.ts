import { transformFileSync } from '@babel/core';
import * as path from 'path';
import plugin from '../src';

describe('babel-plugin-reactive', () => {
  it('works', () => {
    const result = transformFileSync(path.resolve(__dirname, './fixtures/simple.js'), {
      plugins: ['@babel/plugin-transform-react-jsx', [plugin]],
      babelrc: false,
    });
    expect(result!.code).toMatchSnapshot();
  });
});
