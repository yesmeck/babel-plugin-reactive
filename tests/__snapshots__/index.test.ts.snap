// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-reactive works 1`] = `
"import { useState as _useState } from \\"react\\";
import React from 'react';

const App = () => {
  const [count, setcount] = _useState(0);

  return React.createElement(\\"button\\", {
    onClick: () => setcount(_count => _count += 1)
  }, \\"Clicked \\", count, \\" \\", count === 1 ? 'time' : 'times');
};

export default App;"
`;
